### ISSUE TITLE
6.0.1 fails to run 'php artisan optimize' with Unable to prepare route [hardware/{assetId}/checkout] for serialization. Another route has already been assigned name [checkout/hardware].

### ISSUE THREAD

**Issue by davidbuzz** [grokability/snipe-it#11163]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

 $ php artisan optimize
Configuration cache cleared!
Configuration cached successfully!
Route cache cleared!

   LogicException 

  Unable to prepare route [hardware/{assetId}/checkout] for serialization. Another route has already been assigned name [checkout/hardware].

  at vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:218
    214▕             $route->name($name = $this->generateRouteName());
    215▕ 
    216▕             $this->add($route);
    217▕         } elseif (! is_null($symfonyRoutes->get($name))) {
  ➜ 218▕             throw new LogicException("Unable to prepare route [{$route->uri}] for serialization. Another route has already been assigned name [{$name}].");
    219▕         }
    220▕ 
    221▕         $symfonyRoutes->add($route->getName(), $route->toSymfonyRoute());
    222▕ 

      +30 vendor frames 
  31  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()

### Reproduction steps

1. try to run artisan optimize
2.
3.
...

### Expected behavior

it should run without error

### Snipe-IT Version

6.0.1

### Operating System

ubuntu linux 

### Web Server

CLI

### PHP Version

7.4

### Operating System

n/a

### Browser

n/a

### Version

n/a

### Error messages

```shell
$ php artisan optimize
Configuration cache cleared!
Configuration cached successfully!
Route cache cleared!

   LogicException 

  Unable to prepare route [hardware/{assetId}/checkout] for serialization. Another route has already been assigned name [checkout/hardware].

  at vendor/laravel/framework/src/Illuminate/Routing/AbstractRouteCollection.php:218
    214▕             $route->name($name = $this->generateRouteName());
    215▕ 
    216▕             $this->add($route);
    217▕         } elseif (! is_null($symfonyRoutes->get($name))) {
  ➜ 218▕             throw new LogicException("Unable to prepare route [{$route->uri}] for serialization. Another route has already been assigned name [{$name}].");
    219▕         }
    220▕ 
    221▕         $symfonyRoutes->add($route->getName(), $route->toSymfonyRoute());
    222▕ 

      +30 vendor frames 
  31  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
```

**Comment by uberbrady** [comment]
Yeah, we should probably fix this - repeated route names probably are going to eventually break in some version of the framework.

BTW, why are you doing the `optimize` step? Just trying to get things to 'go faster' or is there something specific you're trying to do? (This is more from curiosity than anything else, of course)

**Comment by snipe** [comment]
I already have a branch in process to fix this. It was because of the Great Route Changes of Laravel 8. :( Not hard to do, but LOTS of places reference routes by name, so we have to be careful with testing.

**Comment by uberbrady** [comment]
Well, it sounds like `php artisan optimize` will be a great development tool for that! Hah :) I'll assign it to you then.

**Comment by snipe** [comment]
No, it won't - it will only check to see that the routes can be cached (and there are often lots of valid reasons they can't be, including constructors that do weird things). I mean testing the app, tits to toenails by testing to make sure we're not 500ing - it's going to be a bear.

**Comment by krebbi** [comment]
"clone/license" is also assigned in routes/web.php and in /routes/web/licenses.php

**Comment by snipe** [comment]
It’s a known issue (there are waaaay more than just that one). It’s kinda low priority though, since it’s not necessary to run the optimize command. When benched, it doesn’t offer that much

**Comment by snipe** [comment]
This was fixed quite a while back.
